executor/ddl.go

```diff
@@ -224,6 +238,42 @@ func (e *DDLExec) executeTruncateTable(s *ast.TruncateTableStmt) error {
 	return err
 }
 
+func (e *DDLExec) executeTruncateLocalTemporaryTable(s *ast.TruncateTableStmt) error {
+	dbInfo, exists := e.ctx.GetInfoSchema().(infoschema.InfoSchema).SchemaByName(s.Table.Schema)
+	if !exists {
+		return infoschema.ErrDatabaseNotExists.GenWithStackByArgs(s.Table.Schema)
+	}
+
+	localTempTables := e.getLocalTemporaryTables()
+	if localTempTables == nil {
+		return infoschema.ErrTableNotExists.GenWithStackByArgs(s.Table.Schema, s.Table.Name)
+	}
+
+	tbl, exists := localTempTables.TableByName(s.Table.Schema, s.Table.Name)
+	if !exists {
+		return infoschema.ErrTableNotExists.GenWithStackByArgs(s.Table.Schema, s.Table.Name)
+	}
+
+	tblInfo := tbl.Meta()
+
+	newTbl, err := e.newTemporaryTableFromTableInfo(tblInfo.Clone())
+	if err != nil {
+		return err
+	}
+
+	localTempTables.RemoveTable(s.Table.Schema, s.Table.Name)
+	if err := localTempTables.AddTable(dbInfo, newTbl); err != nil {
+		return err
+	}
```

Comment:
We don't need to remove table and add table, we can just filter out the data of that table?

Reply:
Oh, I see, because of the auto ID allocator?

Reply:
auto ID is one of the reasons. In fact it is the simplest way to 'reset' a table. And it is same with previous practice for dropping a normal table.